src/lightning/data/streaming/dataset.py

```diff
@@ -42,6 +56,7 @@ def __init__(
         drop_last: bool = False,
         seed: int = 42,
         serializers: Optional[Dict[str, Serializer]] = None,
+        checkpoint_interval: int = 60 * 5,
```

Comment:
The caveat here is that 1) the checkpoint might be outdated when we call state_dict() if the interval is too large and 2) the workers may have saved progress for a sample that has not yet been collated into a batch and fetched by the user.

Reply:
Yes. Without patching the DataLoader, this is the best proxy. It won't be perfect but gives a good approximation for a V0.
We can revisit later on.

Reply:
Putting it down to 1 min.